mycity/mycity/intents/location_utils.py

```diff
@@ -0,0 +1,305 @@
+    else:
+        print(error_message)
+        closest_location_info = { feature_type: False,
+                                  DRIVING_DISTANCE_TEXT_KEY: False,
+                                  DRIVING_TIME_TEXT_KEY: False }
+    closest_location_info = \
+        _parse_closest_location_info(feature_type, closest_location_info)
+    
+    return closest_location_info
+
+
+########################################################################
+# Function that interacts with ArcGIS feature servers. Feature servers #
+# serve information about locations in Boston, like locations defined  #
+# as "open spaces" or snow emergency parking lots                      #
+########################################################################
+
+
+def get_features_from_feature_server(url, query):
+    """
+    Given a url to a City of Boston Feature Server, return a list
+    of Features (for example, parking lots that are not full)
+    :param url: url for Feature Server
+    :param query: query to select features (example: "Spaces > 0")
+    :return features: list of all features
+    """
+
+    print(
+        '[method: location_utils.get_features_from_feature_server]',
+        'url received:',
+        url,
+        'query received:',
+        query
+    )
+
+    features = []
+    f = FeatureLayer(url = url)
+    feature_set = f.query(where = query)
+    for feature in feature_set:
+        features.append(feature.as_row[0]) # [0] = data, [1] = column names
+    return features
+
+
+#############################################
+# Functions that interact with Google Maps  #
+#############################################
+
+
+def _get_driving_info(origin, feature_type, destinations):
+    """
+    Gets the driving info from the provided origin address to each destination
+    address
+    :param origin: string containing driving starting address
+    :param destinations: array of address to calculate driving info from origin
+    address
+    :param feature_name: string representing name of this feature
+    :param feature_key: string that identifies type of feature we're getting 
+    directions to 
+    :return driving_infos: dictionary with address, distance, and driving time from origin
+    address
+    """
+    print(
+        '[method: location_utils._get_driving_info]',
+        'origin received:',
+        origin,
+        'feature_type received:',
+        feature_type,
+        'destinations received (printing first five):',
+        destinations[:5],       # only print first five destinations
+        'count(destinations):',
+        len(destinations)
+    )
+
+    url_parameters = _setup_google_maps_query_params(origin, destinations)
+    driving_directions_url = GOOGLE_MAPS_URL
+    driving_infos = None
+    with requests.Session() as session:
+        response = session.get(driving_directions_url, params=url_parameters) # this is failing for open_spaces: problem with params?
```

Comment:
This only fails for the open spaces intent which has been removed...I think all the intents that use arcgis will have to be removed because Amazon spikes the computation if it takes longer than 3.0 ms